pkg/cmd/pulumi/env.go

```diff
@@ -52,7 +51,8 @@ func newEnvCmd() *cobra.Command {
 			}
 			cmdutil.PrintTable(table)
 			if foundError {
-				return result.Error("Invalid environmental variables found")
+				cmdutil.Diag().Errorf(diag.Message("", "Invalid environmental variables found"))
+				return cmdutil.ErrBail
```

Comment:
This looks like a change in behavior. It's an experimental command so it should be fine.